bigchaindb/tendermint/commands.py

```diff
@@ -69,8 +85,17 @@ def start():
 
     setproctitle.setproctitle('bigchaindb')
 
+    # Start the ABCIServer and watch for signals to stop
+    # the server and the relevant child bigchaindb.BIGCHAINDB_PROCESSES.
     app = ABCIServer(app=App())
-    app.run()
+    gevent.signal(signal.SIGTERM, sig_handler, app.server)
```

Comment:
isn't just SIGINT sufficient? why do we need to handle SIGTERM and SIGQUIT?

Reply:
The purpose of handling `SIGTERM`, `SIGQUIT` and `SIGINT`  is to terminate a (parent)process and give it an opportunity to handle graceful shut down, which is the use case, we are trying to address here. So IMO, might as well handle all the signals which allow for `graceful` shutdown.

e.g. We cannot handle `SIGKILL` i.e. `kill -9 <pid>` because it terminates a process immediately and is sort of a last resort. Similarly, `SIGSTOP`. For `SIGKILL` and `SIGSTOP` the user can use the  following i.e.

```
$ pgrep bigchaindb | xargs kill -9 bigchaindb 
```